addons/giltracker/giltracker.lua

```diff
@@ -0,0 +1,182 @@
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'giltracker'
+_addon.author = 'sylandro'
+_addon.version = '1.0.0'
+_addon.language = 'English'
+
+config = require('config')
+images = require('images')
+texts = require('texts')
+
+hideKey = 70
+is_hidden_by_cutscene = false
+is_hidden_by_key = false
+
+defaults = {}
+defaults.HideKey = 70
+defaults.GilText = {}
+defaults.GilText.bg = {}
+defaults.GilText.bg.alpha = 100
+defaults.GilText.bg.red = 0
+defaults.GilText.bg.green = 0
+defaults.GilText.bg.blue = 0
+defaults.GilText.bg.visible = false
+defaults.GilText.text = {}
+defaults.GilText.text.font = 'sans-serif'
+defaults.GilText.text.fonts = {'Arial','Trebuchet MS'}
+defaults.GilText.text.size = 9
+defaults.GilText.flags = {}
+defaults.GilText.flags.italic = true
+defaults.GilText.flags.bold = false
+defaults.GilText.flags.right = true
+defaults.GilText.flags.bottom = true
+defaults.GilText.pos = {}
+defaults.GilText.pos.x = -285
+defaults.GilText.pos.y = -35
+defaults.GilText.text.alpha = 255
+defaults.GilText.text.red = 253
+defaults.GilText.text.green = 252
+defaults.GilText.text.blue = 250
+defaults.GilText.text.stroke = {}
+defaults.GilText.text.stroke.alpha = 200
+defaults.GilText.text.stroke.red = 50
+defaults.GilText.text.stroke.green = 50
+defaults.GilText.text.stroke.blue = 50
+defaults.GilText.text.stroke.width = 2
+defaults.GilText.text.visible = true
+defaults.GilImage = {}
+defaults.GilImage.color = {}
+defaults.GilImage.color.alpha = 255
+defaults.GilImage.color.red = 255
+defaults.GilImage.color.green = 255
+defaults.GilImage.color.blue = 255
+defaults.GilImage.visible = true
+
+local settings = config.load(defaults)
+config.save(settings)
+
+settings.GilImage.texture = {}
+settings.GilImage.texture.path = windower.addon_path..'gil.png'
+settings.GilImage.texture.fit = true
+settings.GilImage.size = {}
+settings.GilImage.size.height = 23
+settings.GilImage.size.width = 23
+settings.GilImage.draggable = false
+settings.GilImage.repeatable = {}
+settings.GilImage.repeatable.x = 1
+settings.GilImage.repeatable.y = 1
+
+gil_image = images.new(settings.GilImage)
+gil_text = texts.new(settings.GilText)
+
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in then
+        initialize()
+    end
+end)
+
+windower.register_event('login', function()
+    initialize()
+end)
+
+windower.register_event('logout', function(...)
+    hide()
+end)
+
+windower.register_event('add item', function(...)
+    update_gil()
```

Comment:
This updates gil every time any item is added. See comments below for `remove item`, the same applies here.